android/src/main/java/com/swmansion/rnscreens/ScreenStackFragment.java

```diff
@@ -232,10 +232,21 @@ public NotifyingCoordinatorLayout(@NonNull Context context, ScreenFragment fragm
     public void startAnimation(Animation animation) {
       // For some reason View##onAnimationEnd doesn't get called for
       // exit transitions so we use this hack.
-      AnimationSet set = new AnimationSet(true);
-      set.addAnimation(animation);
-      set.setAnimationListener(mAnimationListener);
-      super.startAnimation(set);
+      // We also have some animations that are an AnimationSet, so we don't wrap them
+      // in another set since it causes some visual glitches when going forward.
+      // We also set the listener only when going forward, since when going back
+      // there is already a listener for dismiss action, which would be overridden
+      // and also the hack is not necessary when going back since the lifecycle methods
```

Comment:
the word hack has some bad connotations and id prefer not to use it when other words seem more appropriate